plugins/outputs/cloudwatch/util.go

```diff
@@ -46,11 +47,25 @@ const (
 	unitOverheads = 42
 )
 
-func publishJitter(publishInterval time.Duration) (publishJitter time.Duration) {
-	r := rand.New(rand.NewSource(time.Now().Unix()))
-	jitter := r.Int63n(int64(publishInterval.Seconds()))
-	publishJitter = time.Duration(jitter) * time.Second
-	return
+// Global set once.
+var doOnce sync.Once
+var gRand *rand.Rand
+
+// publisJitterInt returns a random int64 between 0 and the given maxVal.
+func publisJitterInt(maxVal int64) int64 {
+	// Set seed once at startup
+	doOnce.Do(func() {
+		gRand = rand.New(rand.NewSource(time.Now().Unix()))
+	})
```

Comment:
In the past when using a `rand` like library, I just initialize/seed once.
I am happy to revert to the old way of always seeding on each call.
It wasn't much of an issue before since the function was called once when the CloudWatch output plugin was initialized.

Reply:
yeah I don't think we stand to really benefit much from creating a new random every time.